src/harness/testcases/WINNF_FT_S_PCR_testcase.py

```diff
@@ -0,0 +1,263 @@
+    for dump_file in response['files']:
+      self.assertContainsRequiredFields("ActivityDumpFile.schema.json", dump_file)
+      downloaded_file = None
+      if dump_file['recordType'] == 'zone':
+        url = str(dump_file['url']).strip()
+        downloaded_file = self._sas.DownloadFile(url)
+        ppa_zone_data.append(downloaded_file)
+
+    # Check if the retrieved FAD PPA zone record matches the PPA ID that was generated
+    # using the admin API.
+    #self.assertEqual(ppa_zone_data[0]['id'], ppa_id)
+
+    uut_ppa_geometry = ppa_zone_data[0]['zone']['features'][0]['geometry']
+    uut_ppa_shapely_geometry = utils.GeoJsonToShapelyGeometry(uut_ppa_geometry)
+
+    # Configure the Census Tract Directory that the PPA uses
+    ppa.ConfigureCensusTractDriver(CONFIG.GetCensusTractsDir())
+
+    # PpaCreationModel requires the input registrationRequests to have 'installationParam'.
+    # But this parameter is removed for devices where conditionals are pre-loaded.
+    # Adding the 'installationParam' into registrationRequests by taking the
+    # corresponding values from conditionalRegistrationData.
+    for device in config['registrationRequests']:
+      if not device.has_key('installationParam'):
+        for conditional_params in config['conditionalRegistrationData'][
+          'registrationData']:
+          # Check if FCC_ID+Serial_Number present in registrationRequest
+          # and conditional_params match and add the 'installationParam'.
+          if conditional_params['fccId'] == device['fccId'] and \
+                          conditional_params['cbsdSerialNumber'] == device['cbsdSerialNumber']:
+            device.update({'installationParam': 
+              conditional_params['installationParam']})
+            device.update({'cbsdCategory': 
+                            conditional_params['cbsdCategory']})
+    # Trigger PPA creation.
+    test_harness_ppa_geometry = ppa.PpaCreationModel(
+        config['registrationRequests'], pal_records)
+    
+    # Convert PPA reference GeoJSON format to Shapely geometry object.
+    test_harness_ppa_shapely_geometry = utils.GeoJsonToShapelyGeometry(
+        json.loads(test_harness_ppa_geometry))
+
+    # Calculates symmetric difference of PPA created by SAS UUT and PPA reference model.
+    symmetric_difference = test_harness_ppa_shapely_geometry.\
+                                    symmetric_difference(uut_ppa_shapely_geometry)
+
+    # Check if the PPA generated by the SAS UUT is fully contained within the service area.
+    self.assertPpaWithinServiceArea(pal_records, uut_ppa_geometry)
+
+    # Check the Equation 8.3.1 in Test Specfification is satisified w.r t [n.12, R2-PAL-05]
+    # Check the area of the non-overlapping difference between
+    # the maximum PPA boundary created by SAS UUT shall be no more than 10% of the area
+    # of the maximum PPA boundary created by the Reference Model.
+    self.assertLess(symmetric_difference.area,
```

Comment:
This is not exactly correct.
Use instead:
```
self.assertTrue(utils.PolygonsAlmostEqual(test_harness_ppa_shapely_geometry, 
                                        uut_ppa_shapely_geometry))
```

Reply:
Updated   accordingly